### PR TITLE
Add releaseVerification option

### PR DESCRIPTION
**Pull request by ryw**
Adds a top level flag to designate what types of deployments are allowed on this Astronomer deployment.

* Moved from astronomer/astronomer/pull/943
* re: [Astronomer Core Action Plan](https://www.notion.so/astronomerio/Astronomer-Core-Action-Plan-fea56f20e06c44138ddb54d9fd230de6) and astronomer/issues#1742

### CODE REVIEW

**Comment by ryw** [comment]
`mode` seemed too generic -- but is `verificationMode` the best name?

**Review by adam2k — APPROVED** [review]
🎉

File: config/default.yaml [hunk 1]
```diff
@@ -398,6 +398,12 @@ deployments:
     # child airflow deployments are kept up to date and on the latest version.
     version: 0.0.0
 
+  # verificationMode determines what docker images are allowed to be installed
+  # STABLE = only official Astronomer Core releases that mirror Airflow releaseVersion
+  # EDGE   = also allow Astronomer Core Edge (astronomer/core:edge) to be used
+  # DEV    = also allow hand-built dev images to be used
+  verificationMode: STABLE
```
Comment (adam2k) [inline config/default.yaml:405]: Some other names:

`releaseChannel`, `releaseMode`, `imageVerification`, `airflowReleaseMode`

Comment (ryw) [reply config/default.yaml:405]: maybe `releaseVerification`

**Comment by ryw** [comment]
Changed to `releaseVerification`

**Pull request merged**